### DISCUSSION TITLE
multiple modal

### DISCUSSION THREAD

**Discussion by malzamri** [chakra-ui/chakra-ui#3378] [Q&A] [answered]
how to use multiple modal or alert in a single component?

**Comment by dodas** [comment, marked as answer]
Hey @malzamri,

I guess you have problem understanding how `useDisclosure` works.
`useDisclosure` just returns object with `isOpen` boolean and functions to update it: `onClose` and `onOpen`. [[See the docs]](https://chakra-ui.com/docs/hooks/use-disclosure)
You can use it multiple times in a component, you just have to rename the variables.

I think it is more readable without object destructuring, like this:
```js
function MyComponent() {
  const modal1 = useDisclosure()
  const modal2 = useDisclosure()
  
  return (
    <>
      <Button onClick={modal1.onOpen}>Open Modal 1</Button>
      <Button onClick={modal2.onOpen}>Open Modal 2</Button>

      <Modal isOpen={modal1.isOpen} onClose={modal1.onClose}>
        // omitted for brevity
      </Modal>
      
      <Modal isOpen={modal2.isOpen} onClose={modal2.onClose}>
        // omitted for brevity
      </Modal>
    </>
  )
}
```

**Reply by malzamri** [reply]
Ohhh, thank you so much

**Reply by lautarojgarcia177** [reply]
Had the same doubt. Thank you!

**Reply by pankajpareek026** [reply]
Thank you !
